src/proto/streams/send.rs

```diff
@@ -107,6 +107,17 @@ impl<B> Send<B> where B: Buf {
         stream.state.send_close()
     }
 
+    pub fn send_reset(&mut self, reason: Reason,
+                      stream: &mut store::Ptr<B>)
+        -> Result<(), ConnectionError>
+    {
+        stream.state.send_reset(reason)?;
```

Comment:
It looks like the reset will still be sent if the stream state is already closed (potentially reset already). Is this expected?